### ISSUE TITLE
request: add support for browsing QML-based websites.

### ISSUE THREAD

**Issue by trusktr** [liri-project/liri-browser#217]
If the site serves text/qml instead of text/html, render with the QML engine instead of Chromium's HTML engine.

Someone needs to do this! I think Liri is a nice project to make it happen...

**Comment by tim-sueberkrueb** [comment]
This is a nice suggestion, we thought about it already. 
Though, for security reasons, this would require to refactor a lot of code.
Web views would have to run in separate QML engines to ensure the QML-based websites
have absolutely no access to the system and browser settings.

**Comment by trusktr** [comment]
That's true! Maybe we can open a request on Qt for some type of QML sandbox that has no system access by default, and those settings could be configurable? 

Then someone making a browser, you here at Liri, could implement permissions systems, etc.

**Comment by tim-sueberkrueb** [comment]
I think it's not necessary to make an upstream request for this. It's already possible to create our own sandbox by creating separate QML Views and engines. By setting an empty list as `importPathList` we can deny access to any modules and only expose things that should be exposed.
